R/combine_mixed_crops.R

```diff
@@ -0,0 +1,123 @@
+#' Combine two-PFT outputs to represent mixed cropping systems
+#'
+#' Rules for combining woody and annual PFT outputs to represent a
+#' mixed cropping system. Supports two methods:
+#'  - "weighted": PFTs partition area (woody_cover + annual_cover = 1) and the
+#'    output is a weighted average: `woody_cover * woody_value + annual_cover * annual_value`.
+#'  - "incremental": preserves the full-area woody baseline (requires `woody_cover == 1`)
+#'    and treats annual as an increment relative to an annual initial baseline:
+#'    `annual_delta = annual_value - annual_init`; `result = woody_value + annual_cover * annual_delta`.
+#'
+#' Vectorization & recycling are handled via `vctrs::vec_recycle_common()`. All scalar
+#' inputs are broadcast to the common length. Length mismatches other than 1 or the
+#' common size trigger a severe error.
+#'
+#' Validation rules (severe errors unless otherwise noted):
+#'  * No input values may be NA (including covers, pool sizes, annual_init if required)
+#'  * Covers must lie within [0,1]
+#'  * Method "incremental": `woody_cover` must be 1 (within 0.1%); if not, severe error
+#'  * Method "incremental": `annual_init` required
+#'  * Method "weighted": rows whose `woody_cover + annual_cover` differ from 1 by more than tolerance
+#'    are set to NA in the result; a single aggregated warning is emitted listing the count
+#'
+#' @param woody_value numeric. Pool size for the woody PFT (kg/m2).
+#' @param annual_value numeric. Pool size for the annual PFT (kg/m2).
+#' @param annual_init numeric, required for method = "incremental"; the initial annual pool.
+#' @param annual_cover numeric. Fractional cover of the annual PFT (f_annual, 0-1).
+#' @param woody_cover numeric. Fractional cover of the woody PFT (f_woody, 0-1). Must be 1 for incremental.
+#' @param method character. One of "weighted" or "incremental".
+#'
+#' @return numeric vector of combined values.
+#'
+#' @examples
+#' # Discrete mixing (weights sum to 1)
+#' combine_mixed_crops(
+#'   woody_value = 100, annual_value = 50,
+#'   annual_cover = 0.2, woody_cover = 0.8, method = "weighted"
+#' )
+#'
+#' # Overlap: preserve woody baseline (woody_cover==1), add annual increment scaled by cover
+#' combine_mixed_crops(
+#'   woody_value = 200, annual_value = 220, annual_init = 200,
+#'   annual_cover = 0.3, woody_cover = 1.0, method = "incremental"
+#' )
+combine_mixed_crops <- function(woody_value,
+                                annual_value,
+                                annual_init = NULL,
+                                annual_cover,
+                                woody_cover,
```

Comment:
Values with defaults go at end of args
```suggestion
                                annual_cover,
                                woody_cover,
                                annual_init = NULL,
```